src/Cake/Program.cs

```diff
@@ -36,90 +23,27 @@ public static int Main()
-            builder.RegisterType<CakeEnvironment>().As<ICakeEnvironment>().SingleInstance();
-            builder.RegisterType<CakeArguments>().As<ICakeArguments>().SingleInstance();
-            builder.RegisterType<Globber>().As<IGlobber>().SingleInstance();
-            builder.RegisterType<ProcessRunner>().As<IProcessRunner>().SingleInstance();
-            builder.RegisterType<ScriptAliasFinder>().As<IScriptAliasFinder>().SingleInstance();
-            builder.RegisterType<CakeReportPrinter>().As<ICakeReportPrinter>().SingleInstance();
-            builder.RegisterType<CakeConsole>().As<IConsole>().SingleInstance();
-            builder.RegisterType<ScriptAnalyzer>().As<IScriptAnalyzer>().SingleInstance();
-            builder.RegisterType<ScriptProcessor>().As<IScriptProcessor>().SingleInstance();
-            builder.RegisterType<ScriptConventions>().As<IScriptConventions>().SingleInstance();
-            builder.RegisterType<NuGetToolResolver>().As<INuGetToolResolver>().SingleInstance();
-            builder.RegisterType<WindowsRegistry>().As<IRegistry>().SingleInstance();
-            builder.RegisterType<CakeContext>().As<ICakeContext>().SingleInstance();
-
-            // NuGet addins support
-            builder.RegisterType<NuGetVersionUtilityAdapter>().As<INuGetFrameworkCompatibilityFilter>().As<IFrameworkNameParser>().SingleInstance();
-            builder.RegisterType<NuGetPackageAssembliesLocator>().As<INuGetPackageAssembliesLocator>().SingleInstance();
-            builder.RegisterType<NuGetPackageReferenceBundler>().As<INuGetPackageReferenceBundler>().SingleInstance();
-            builder.RegisterType<NuGetAssemblyCompatibilityFilter>().As<INuGetAssemblyCompatibilityFilter>().SingleInstance();
-            builder.RegisterType<AssemblyFrameworkNameParser>().As<IAssemblyFrameworkNameParser>().SingleInstance();
-
-            // URI resource support.
-            builder.RegisterType<NuGetPackageInstaller>().As<IPackageInstaller>().SingleInstance();
-            builder.RegisterType<NuGetPackageContentResolver>().As<INuGetPackageContentResolver>().SingleInstance();
-
-            if (mono)
+            // Build the container.
+            using (var container = builder.Build())
             {
-                // Mono scripting.
-                builder.RegisterType<Scripting.Mono.MonoScriptEngine>().As<IScriptEngine>().SingleInstance();
-            }
-            else
-            {
-                // Roslyn related services.
-                builder.RegisterType<RoslynScriptEngine>().As<IScriptEngine>().SingleInstance();
-                builder.RegisterType<RoslynScriptSessionFactory>().SingleInstance();
-                builder.RegisterType<RoslynNightlyScriptSessionFactory>().SingleInstance();
-            }
+                // Parse the options.
+                var parser = container.Resolve<IArgumentParser>();
+                var options = parser.Parse(args) ?? new CakeOptions();
```

Comment:
This should be within a try catch, i.e. an invalid verbosity will now generate an unhandled exception
```
Unhandled Exception: System.InvalidOperationException: The value 'Invalid' is not a valid verbosity.
   at Cake.Arguments.VerbosityParser.Parse(String value) in \src\Cake\Arguments\VerbosityParser.cs:line 49
   at Cake.Arguments.ArgumentParser.ParseOption(String name, String value, CakeOptions options) in \src\Cake\Arguments\ArgumentParser.cs:line 1
30
   at Cake.Arguments.ArgumentParser.Parse(IEnumerable`1 args) in \src\Cake\Arguments\ArgumentParser.cs:line 46
   at Cake.Program.Main() in \src\Cake\Program.cs:line 34
```

Where as it before would just generate an error:
```
Error: The value 'Invalid' is not a valid verbosity.
```